### DISCUSSION TITLE
Add private key based authentication

### DISCUSSION THREAD

**Discussion by krzyk** [ankitpokhrel/jira-cli#603] [Ideas]
Please add private key based authentication (one that works e.g. from curl by passing in the `--cert` and `--key` options).

**Comment by markhatch** [comment]
I have been implementing this in a fork. I added mtls cert support to local installs. I can try to open a PR with the feature, but I likely will need to clean it up a bit.

**Reply by markhatch** [reply]
@ankitpokhrel would you be open to that? Essentially I add a authType `mtls` which then prompts users to enter CA Cert + Client Certificates file locations, which get used instead of basic/bearer auth. It works on my on-prem server.

**Reply by ankitpokhrel** [reply]
Hey, would absolutely love and appreciate your pull request! :)

**Reply by markhatch** [reply]
Took a pass at implementation here: https://github.com/ankitpokhrel/jira-cli/pull/615
